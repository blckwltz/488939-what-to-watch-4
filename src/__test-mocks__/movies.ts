import {Movie} from '../types/movie';

const featuredMovie: Movie = {
  id: 1,
  isFavorite: false,
  title: `Pulp Fiction`,
  genre: `Drama`,
  releaseDate: 1994,
  runTime: 130,
  cover: `img/the-grand-budapest-hotel-poster.jpg`,
  poster: `img/pulp-fiction.jpg`,
  backgroundImage: `img/the-grand-budapest-hotel-poster.jpg`,
  backgroundColor: `#ffffff`,
  videoSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  previewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  rating: {
    score: 8.9,
    count: 578,
  },
  description: `The lives of two mob hitmen, a boxer, a gangster and his wife, and a pair of diner bandits
      intertwine in four tales of violence and redemption.`,
  director: `Quentin Tarantino`,
  cast: [`John Travolta`, `Uma Thurman`, `Samuel L. Jackson`],
};

const moviesList: Movie[] = [
  {
    id: 1,
    isFavorite: false,
    title: `Macbeth`,
    genre: `Drama`,
    releaseDate: 2015,
    runTime: 120,
    cover: `img/the-grand-budapest-hotel-poster.jpg`,
    poster: `img/macbeth.jpg`,
    backgroundImage: `img/the-grand-budapest-hotel-poster.jpg`,
    backgroundColor: `#ffffff`,
    videoSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    rating: {
      score: 6.6,
      count: 120,
    },
    description: `Macbeth, the Thane of Glamis, receives a prophecy from a trio of witches that one day he will
      become King of Scotland. Consumed by ambition and spurred to action by his wife, Macbeth murders his king
      and takes the throne for himself.`,
    director: `Justin Kurzel`,
    cast: [`Michael Fassbender`, `Marion Cotillard`, `Jack Madigan`],
  },
  {
    id: 2,
    isFavorite: false,
    title: `We need to talk about Kevin`,
    genre: `Drama`,
    releaseDate: 2011,
    runTime: 100,
    cover: `img/the-grand-budapest-hotel-poster.jpg`,
    poster: `img/we-need-to-talk-about-kevin.jpg`,
    backgroundImage: `img/the-grand-budapest-hotel-poster.jpg`,
    backgroundColor: `#ffffff`,
    videoSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    rating: {
      score: 7.5,
      count: 134,
    },
    description: `Kevin's mother struggles to love her strange child, despite the increasingly dangerous things
      he says and does as he grows up. But Kevin is just getting started, and his final act will be beyond
      anything anyone imagined.`,
    director: `Lynne Ramsay`,
    cast: [`Tilda Swinton`, `John C. Reilly`, `Ezra Miller`],
  },
  {
    id: 3,
    isFavorite: false,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    genre: `Adventure`,
    releaseDate: 2018,
    runTime: 185,
    cover: `img/the-grand-budapest-hotel-poster.jpg`,
    poster: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    backgroundImage: `img/the-grand-budapest-hotel-poster.jpg`,
    backgroundColor: `#ffffff`,
    videoSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    rating: {
      score: 6.6,
      count: 300,
    },
    description: `The second installment of the "Fantastic Beasts" series featuring the adventures
      of Magizoologist Newt Scamander.`,
    director: `David Yates`,
    cast: [`Eddie Redmayne`, `Katherine Waterston`, `Dan Fogler`],
  },
  {
    id: 4,
    isFavorite: false,
    title: `Shutter Island`,
    genre: `Mystery`,
    releaseDate: 2010,
    runTime: 154,
    cover: `img/the-grand-budapest-hotel-poster.jpg`,
    poster: `img/shutter-island.jpg`,
    backgroundImage: `img/the-grand-budapest-hotel-poster.jpg`,
    backgroundColor: `#ffffff`,
    videoSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    rating: {
      score: 8.1,
      count: 257,
    },
    description: `In 1954, a U.S. Marshal investigates the disappearance of a murderer who escaped from a hospital
      for the criminally insane.`,
    director: `Martin Scorsese`,
    cast: [`Leonardo DiCaprio`, `Emily Mortimer`, `Mark Ruffalo`],
  },
  {
    id: 5,
    isFavorite: false,
    title: `Snatch`,
    genre: `Comedy`,
    releaseDate: 2000,
    runTime: 107,
    cover: `img/the-grand-budapest-hotel-poster.jpg`,
    poster: `img/snatch.jpg`,
    backgroundImage: `img/the-grand-budapest-hotel-poster.jpg`,
    backgroundColor: `#ffffff`,
    videoSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    rating: {
      score: 8.3,
      count: 300,
    },
    description: `Unscrupulous boxing promoters, violent bookmakers, a Russian gangster, incompetent amateur robbers
      and supposedly Jewish jewelers fight to track down a priceless stolen diamond.`,
    director: `Guy Ritchie`,
    cast: [`Jason Statham`, `Brad Pitt`, `Benicio Del Toro`],
  },
  {
    id: 6,
    isFavorite: false,
    title: `Pulp Fiction`,
    genre: `Drama`,
    releaseDate: 1994,
    runTime: 130,
    cover: `img/the-grand-budapest-hotel-poster.jpg`,
    poster: `img/pulp-fiction.jpg`,
    backgroundImage: `img/the-grand-budapest-hotel-poster.jpg`,
    backgroundColor: `#ffffff`,
    videoSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    rating: {
      score: 8.9,
      count: 578,
    },
    description: `The lives of two mob hitmen, a boxer, a gangster and his wife, and a pair of diner bandits
      intertwine in four tales of violence and redemption.`,
    director: `Quentin Tarantino`,
    cast: [`John Travolta`, `Uma Thurman`, `Samuel L. Jackson`],
  },
  {
    id: 7,
    isFavorite: false,
    title: `Revenant`,
    genre: `Adventure`,
    releaseDate: 2015,
    runTime: 120,
    cover: `img/the-grand-budapest-hotel-poster.jpg`,
    poster: `img/revenant.jpg`,
    backgroundImage: `img/the-grand-budapest-hotel-poster.jpg`,
    backgroundColor: `#ffffff`,
    videoSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    rating: {
      score: 8.0,
      count: 1001,
    },
    description: `A frontiersman on a fur trading expedition in the 1820s fights for survival after being
      mauled by a bear and left for dead by members of his own hunting team.`,
    director: `Alejandro G. Iñárritu`,
    cast: [`Leonardo DiCaprio`, `Tom Hardy`, `Will Poulter`],
  },
  {
    id: 8,
    isFavorite: false,
    title: `Bohemian Rhapsody`,
    genre: `Biography`,
    releaseDate: 2018,
    runTime: 170,
    cover: `img/the-grand-budapest-hotel-poster.jpg`,
    poster: `img/bohemian-rhapsody.jpg`,
    backgroundImage: `img/the-grand-budapest-hotel-poster.jpg`,
    backgroundColor: `#ffffff`,
    videoSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    rating: {
      score: 8.0,
      count: 240,
    },
    description: `The story of the legendary British rock band Queen and lead singer Freddie Mercury,
      leading up to their famous performance at Live Aid (1985).`,
    director: `Bryan Singer`,
    cast: [`Rami Malek`, `Lucy Boynton`, `Gwilym Lee`],
  },
];

export {featuredMovie, moviesList};
