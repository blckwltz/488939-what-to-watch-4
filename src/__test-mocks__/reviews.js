const reviews = [
  {
    id: 1,
    author: {
      id: 1,
      name: `Kate Muir`,
    },
    text: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious
          Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    date: `2016-12-24`,
    rating: 8.7,
  },
  {
    id: 2,
    author: {
      id: 2,
      name: `Bill Goodykoontz`,
    },
    text: `I didn't find it amusing, and while I can appreciate the creativity, it's an hour and 40 minutes
          I wish I could take back.`,
    date: `2015-11-18`,
    rating: 5.3,
  },
];

export {reviews};
