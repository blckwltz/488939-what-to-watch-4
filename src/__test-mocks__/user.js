const authorizationStatus = `200`;

export {authorizationStatus};
