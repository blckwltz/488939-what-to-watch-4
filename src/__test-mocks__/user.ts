const authorizationStatus: string = `200`;

export {authorizationStatus};
