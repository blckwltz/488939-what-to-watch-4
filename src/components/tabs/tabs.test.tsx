import React, {ReactElement} from 'react';
import renderer from 'react-test-renderer';
import {noop} from '../../__test-mocks__/noop';
import Tabs from './tabs';


const children: ReactElement[] = [
  <div key="child" title="child" className="child-component">
    <div className="nested-component"/>
  </div>
];

it(`Should render Tabs component correctly`, () => {
  const tree = renderer
    .create(
        <Tabs
          activeItem={0}
          onActiveItemChange={noop}
        >
          {children}
        </Tabs>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
