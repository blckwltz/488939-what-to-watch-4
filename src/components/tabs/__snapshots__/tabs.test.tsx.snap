// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Tabs component correctly 1`] = `
<div
  className="movie-card__desc"
>
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item movie-nav__item--active"
        onClick={[Function]}
      >
        <a
          className="movie-nav__link"
          href="#"
        >
          child
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="nested-component"
  />
</div>
`;
